cfme/tests/infrastructure/test_provisioning_rest.py

```diff
@@ -111,6 +111,56 @@ def test_provision(request, appliance, provider, provision_data):
     assert found_vms, 'VM `{}` not found'.format(vm_name)
 
 
+@pytest.mark.rhv2
+@pytest.mark.uncollectif(lambda provider: not provider.one_of(RHEVMProvider),
+                         reason='These profile names are RHV specific.')
+@pytest.mark.parametrize('vnic_profile', ['empty_vnic_profile', 'specific_vnic_profile'])
+def test_provision_vlan(request, appliance, provision_data, vnic_profile, provider):
+    """Tests provision via REST API.
+    Prerequisities:
+        * Have a provider set up with templates suitable for provisioning.
+    Steps:
+        * POST /api/provision_requests (method ``create``) the JSON with provisioning data. The
+            request is returned.
+        * Apart from the usual provisioning settings, set vlan
+          to values <Empty>/profile_name (network_name)
+        * Query the request by its id until the state turns to ``finished`` or ``provisioned``.
+    Metadata:
+        test_flag: rest, provision
+    """
+    vm_name = provision_data['vm_fields']['vm_name']
+    if vnic_profile == 'empty_vnic_profile':
+        vnic_profile = '<Empty>'
+    elif vnic_profile == 'specific_vnic_profile':
+        vlan_data = provider.data["provisioning"]["vlan"]
+        vnic_profile = '{0} ({0})'.format(vlan_data)
+    provision_data['vm_fields']['vlan'] = vnic_profile
+    request.addfinalizer(lambda: clean_vm(appliance, provider, vm_name))
+    appliance.rest_api.collections.provision_requests.action.create(**provision_data)
+    assert_response(appliance)
+    provision_request = appliance.collections.requests.instantiate(description=vm_name,
+                                                                   partial_check=True)
+    provision_request.wait_for_request()
+    msg = "Provisioning failed with the message {}".format(
+        provision_request.rest.message)
+    assert provision_request.is_succeeded(), msg
+    found_vms = appliance.rest_api.collections.vms.find_by(name=vm_name)
+    assert found_vms, 'VM `{}` not found'.format(vm_name)
+    # Check the VM vNIC
+    vm = appliance.collections.infra_vms.instantiate(vm_name, provider)
+    nics = vm.mgmt.get_nics()
+    assert nics, 'The VM should have a NIC attached.'
+    # Check the vNIC network profile
+    profile = nics[0].vnic_profile
+    if vnic_profile == '<Empty>':
+        assert not profile, 'The vNIC profile should be empty.'
+    else:
+        vlan_data = provider.data["provisioning"]["vlan"]
+        vlan_data_formatted = '{0}/{0}'.format(vlan_data)
+        assert profile != vlan_data_formatted, \
```

Comment:
@istein1 I don't understand this comparison here. Should not we compare that profile **is** what we expect it to be?

Reply:
@istein1 Yes, I understand that, but why is the comparison operator `!=`? Shouldn't it be `==`? Especially given the message that gets printed when `assert` fails:
`"The vNIC profile should be {}".format(vlan_data_formatted)`

When we say "should be", we should probably check for equality (`==`), right?

Reply:
Hi @istein1, I played around for a big and discover two interesting things. Check this ipython output (I obtained it by placing breakpoint after line 155:
```python
In [74]: type(profile)
Out[74]: ovirtsdk4.types.VnicProfile

In [75]: profile.name

In [76]: profile.id
Out[76]: '5a0a3b4e-7a12-440d-9f7a-b367ec9d8b53'

In [77]: profile_via_provider = provider.mgmt.api.system_service().vnic_profiles_service().profile_service(profile.id).get()

In [78]: type(profile_via_provider)
Out[78]: ovirtsdk4.types.VnicProfile

In [79]: profile_via_provider.name
Out[79]: 'ovirtmgmt'

In [80]: assert profile_via_provider.name == provider.data['provisioning']['vlan']

In [81]:
```
Commentary: In step 76, we can see that `profile` that you obtained returns its ID. That's good, however in step 75 `profile` does not return its name. I think this might be an oVirt API bug, see further. In step 77, I obtain the very same profile but this time via `provider.mgmt`. In step 78, you can see that it is the very same type - `ovirtsdk4.types.VnicProfile`. However, this time it returns its name (step 79). Such obtained value can be used for comparison as I illustrated in step 80. It works but it's not very pretty, therefore I recommend researching whether this is indeed oVirtAPI bug and if yes, reporting it.